client/src/service/round/round.test.ts

```diff
@@ -0,0 +1,1773 @@
+import { Card, Stat } from "../../types/card/card.types";
```

Comment:
Wow this is a lot - well done! I haven't had a chance to read it but from  spinning up the branch and viewing the browser it looks like you covered a lot here.

Reply:
I've actually been giving it some thought and I think some of these could just be inlined with the setGame functions in App.tsx. I think I got caught up in the idea that the functions had to be in the service folder, but anything that's manipulating state needs to be in App.tsx, and some of the functions here can do it directly rather than being seperate functions that return a new object. I might refactor slightly to show that if there's time. Which is a bit annoying, because it means I wasted ages writing tests for functions that we won't be able to test anymore, but oh well.